### PR TITLE
Respond with 404 on fetching default credential

### PR DESCRIPTION
**Pull request by marians**
Epic: https://github.com/giantswarm/giantswarm/issues/2652

Similar to https://github.com/giantswarm/credentiald/pull/18, for the searcher service

### CODE REVIEW

File: service/searcher/service.go [hunk 1]
```diff
@@ -61,6 +61,12 @@ func New(config Config) (*Service, error) {
 func (c *Service) Search(request Request) (*Response, error) {
 	c.logger.Log("level", "debug", "message", fmt.Sprintf("searching secret for organization %s, ID %s", request.Organization, request.ID))
 
+	// We never expose the credential-default secret. From the outside, this does not exist.
+	if request.ID == "default" {
```
Comment (JosephSalisbury) [inline service/searcher/service.go]: I'd like `default` to be a constant || pulled from some configuration.

Comment (marians) [reply service/searcher/service.go]: done

File: service/searcher/service.go [hunk 2]
```diff
@@ -61,6 +64,12 @@ func New(config Config) (*Service, error) {
 func (c *Service) Search(request Request) (*Response, error) {
 	c.logger.Log("level", "debug", "message", fmt.Sprintf("searching secret for organization %s, ID %s", request.Organization, request.ID))
 
+	// We never expose the credential-default secret. From the outside, this does not exist.
+	if request.ID == defaultCredentialNameIDPart {
+		c.logger.Log("level", "warn", "message", "attempt to get default credential. Denied.")
```
Comment (kopiczko) [inline service/searcher/service.go]: Should we have a log here? Why?
I think it should be a metric if anything, but I don't see it being useful anyway.

Comment (JosephSalisbury) [reply service/searcher/service.go]: No strong opinion. I don't think this is a particularly important event.

Comment (marians) [reply service/searcher/service.go]: You're right. Removed it.

**Pull request merged**